uxarray/grid/coordinates.py

```diff
@@ -83,40 +88,75 @@ def node_xyz_to_lonlat_rad(node_coord):
     return [d_lon_rad, d_lat_rad]
 
 
-@njit(cache=ENABLE_JIT_CACHE)
-def normalize_in_place(node):
-    """Helper function to project an arbitrary node in 3D coordinates [x, y, z]
-    on the unit sphere. It uses the `np.linalg.norm` internally to calculate
-    the magnitude.
+#@njit(cache=ENABLE_JIT_CACHE)
+def node_xyz_to_lonlat_rad(coords):
+    coords_norm = normalize_in_place(coords)
+    dx = coords_norm[:, 0]
+    dy = coords_norm[:, 1]
+    dz = coords_norm[:, 2]
 
-    Parameters
-    ----------
-    node: float list
-        3D Cartesian Coordinates [x, y, z]
+    dx /= np.absolute(dx * dx + dy * dy + dz * dz)
+    dy /= np.absolute(dx * dx + dy * dy + dz * dz)
+    dz /= np.absolute(dx * dx + dy * dy + dz * dz)
 
-    Returns
-    ----------
-    float list
-        the result unit vector [x, y, z] where :math:`x^2 + y^2 + z^2 = 1`
+    if np.absolute(dz) < (1.0 - ERROR_TOLERANCE):
+        d_lon_rad = np.atan2(dy, dx)
+        d_lat_rad = np.arcsin(dz)
 
-    Raises
-    ----------
-    RuntimeError
-        The input array doesn't have the size of 3.
-    """
-    if len(node) != 3:
-        raise RuntimeError("Input array should have a length of 3: [x, y, z]")
-    return list(np.array(node) / np.linalg.norm(np.array(node), ord=2))
+        if d_lon_rad < 0.0:
+            d_lon_rad += 2.0 * np.pi
+    elif dz > 0.0:
+        d_lon_rad = 0.0
+        d_lat_rad = 0.5 * np.pi
+    else:
+        d_lon_rad = 0.0
+        d_lat_rad = -0.5 * np.pi
+
+    return [d_lon_rad, d_lat_rad]
+
+
+# @njit(cache=ENABLE_JIT_CACHE)
+# def normalize_in_place(node):
+#     """Helper function to project an arbitrary node in 3D coordinates [x, y, z]
+#     on the unit sphere. It uses the `np.linalg.norm` internally to calculate
+#     the magnitude.
+#
+#     Parameters
+#     ----------
+#     node: float list
+#         3D Cartesian Coordinates [x, y, z]
+#
+#     Returns
+#     ----------
+#     float list
+#         the result unit vector [x, y, z] where :math:`x^2 + y^2 + z^2 = 1`
+#
+#     Raises
+#     ----------
+#     RuntimeError
+#         The input array doesn't have the size of 3.
+#     """
+#     if len(node) != 3:
+#         raise RuntimeError("Input array should have a length of 3: [x, y, z]")
+#     return list(np.array(node) / np.linalg.norm(np.array(node), ord=2))
+
+
+#@njit(cache=ENABLE_JIT_CACHE)
+def normalize_in_place(coords):
+
+    coords_norm = coords / np.linalg.norm(coords, ord=2, axis=1)
```

Comment:
I have started a branch with similar WIP code - [here](https://github.com/UXARRAY/uxarray/commit/e4226b902065b5cd33f406ec19bd98719d4aaa69#diff-cdc8fc5f4b48af740c90192c6c361e0fe033e6a7157ff6ee67de9e19b6a99a7dR114)

Reply:
I'll take a look at this! This branch right here is just a few ideas tossed together for now :)